sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionState.scala

```diff
@@ -223,6 +223,7 @@ private[hive] class HiveSessionState(ctx: SQLContext) extends SessionState(ctx)
     conf.getConf(HiveUtils.HIVE_THRIFT_SERVER_ASYNC)
   }
 
+  // TODO: why do we get this from SparkConf but not SQLConf?
```

Comment:
NVM. `spark.sql.hive.thriftServer.singleSession` is crossing `SessionState`. We should move this to `HiveSharedState`/`SharedState`. Thus, it is not right to move it to `SQLConf`.